TM1637TinyDisplay.cpp

```diff
@@ -172,14 +172,16 @@ TM1637TinyDisplay::TM1637TinyDisplay(uint8_t pinClk, uint8_t pinDIO, unsigned in
   m_scrollDelay = scrollDelay;
   // Flip 
   m_flipDisplay = flip;
-  
```

Comment:
While I'm keen to keep progmem space low, I do think to honor the contract of a constructor, we need to initialize the the RAM space we hold for the display shadow memory. This was originally handled with the call to clear() but we could just put it here. 

```
memset(digits,0,sizeof(digits));
```

Thoughts?

Reply:
I always expect variables (memory) to be "0" pre-initialized, but also I always doubt so... **as you wish** 👌

Reply:
I tend to be quite minimalist and avoid writing any code *just in case*, and hand-over those cases (if necessary) to the final programmer.

Reply:
That's a good point. I'm trying to think of any halo effects that would occur if it is not initialized. 
* If the user were to call `flipDisplay()` before `clear()` then the results would be unknown, but I think that is valid as we don't even know what state the display would be in at this point.  
* If they were to call something like `setSegments()` or `showNumber()` with a position argument, any segment not already set would be in an unknown state and may display random elements. Again, new users of the library would know to call `begin()` but that wouldn't help unless they also called `clear()`. 
Previous projects could have some slight impact if they were expecting the constructor to zero out the display memory.

As I'm thinking through this, I'm good with keeping the memory init out of the constructor. However, it suddenly occurs to me... We now require the user to call both `begin()` and `clear()` for every project.  When would you ever want to call `begin()` without a `clear()`?  I can't think of any reason. Why wouldn't we put `clear()` back in to `begin()`?  Am I missing something?  

Full transparency, I'm also a reliability engineer, so I spend too much time in the "just in case" domain. 😁  Thanks for keeping me in check. We want as simple as we can be, but it should still be intuitive.

Reply:
Good points here. It's always tricky to find the right balance and I'm OK with your fix 👌